test/unit/exception_handling_test.rb

```diff
@@ -180,6 +180,20 @@ class SmtpClientErrbackStub < SmtpClientStub
       end
     end
 
+    context "#write_exception_to_log" do
```

Comment:
I'm surprised this isn't a private method.. at any rate though, this ticket is focused on the `log_warning` interface, so at the very least a new test should be added that proves when calling `log_warning`, that the `logged_excluding_reload_filter.last[:severity]` is "WARN" (and similar iwth calling `log_error` having the severity still be FATAL. 

Given this method isn't private, you could leave these lower level unit tests too.

Reply:
I was also surprised by this, but it looks like it was a workaround so ExceptionCatalog#find doesn't get into an infinite loop by calling log_error (which calls ExceptionCatalog#find).